### PR TITLE
Collect Service Credentials and Service Credential Types

### PR DESCRIPTION
**Pull request by lpichler**
3. Collect Service Credentials and Service Credential Type
from https://github.com/RedHatInsights/topological_inventory-core/issues/168 (models: https://github.com/RedHatInsights/topological_inventory-core/pull/170)

@miq-bot assign @slemrmartin

### CODE REVIEW

File: lib/topological_inventory/ansible_tower/parser/service_credential.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+module TopologicalInventory::AnsibleTower
+  class Parser
+    module ServiceCredential
+      def parse_service_credential(credential)
+        service_credential_type = lazy_find(:service_credential_types, :source_ref => credential.credential_type_id.to_s) if credential.respond_to?(:credential_type_id)
+
+        collections.service_credentials.build(
+          parse_base_item(credential).merge(
+            :source_ref              => credential.id.to_s,
+            :source_updated_at       => credential.created,
```
Comment (slemrmartin) [inline lib/topological_inventory/ansible_tower/parser/service_credential.rb]: @lpichler did you mean `:source_created_at`? You can delete this line, it's part of `parse_base_item`

File: lib/topological_inventory/ansible_tower/parser/service_credential.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+            :source_updated_at       => credential.modified,
```
Comment (slemrmartin) [inline lib/topological_inventory/ansible_tower/parser/service_credential.rb]: We're not collecting `source_updated_at` for any other entity type, so I'm for consistency - either always or nowhere. 
So I would prefer to skip it now and if we'll need it, then add it to all models

File: lib/topological_inventory/ansible_tower/parser/service_credential_type.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+module TopologicalInventory::AnsibleTower
+  class Parser
+    module ServiceCredentialType
+      def parse_service_credential_type(credential_type)
+        collections.service_credential_types.build(
+          parse_base_item(credential_type).merge(
+            :source_ref        => credential_type.id.to_s,
+            :source_created_at => credential_type.created,
+            :source_updated_at => credential_type.modified,
```
Comment (slemrmartin) [inline lib/topological_inventory/ansible_tower/parser/service_credential_type.rb]: The same about `source_created|updated_at` as my comments for `ServiceCredential`

**Review by slemrmartin — APPROVED** [review]
:+1: LGTM

**Comment by slemrmartin** [comment]
this PR waits for updated openapi and ingress-api-client PRs before merge

**Comment by lpichler** [comment]
> this PR waits for updated openapi and ingress-api-client PRs before merge

@slemrmartin  can you merge last schema pr ? https://github.com/RedHatInsights/topological_inventory-core/pull/173 

it is included in those PRs:
https://github.com/RedHatInsights/topological_inventory-ingress_api/pull/80
https://github.com/RedHatInsights/topological_inventory-ingress_api-client-ruby/pull/63

**Review by slemrmartin — APPROVED** [review]
:+1: LGTM

**Pull request merged**